### ISSUE TITLE
Allow List Syntax in Attention Block

### ISSUE THREAD

**Issue by k-macmillan** [department-of-veterans-affairs/notification-utils#181]
## User Story - Business Need
Services would like to use a list in "personalisation" to be within a code/attention block. The use of lists in this way is not currently supported according to Portal usage, but we do have API support to allow a list passed in the request to become a bulleted list in the email. This feature was not 100% ready for services to use, this is to close the loop and make it Production ready.

- [ ] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)

**As a** service using VA Notify
**I want** to use a list to generated a bulleted list within a code block
**So that** I can simplify templates

### Additional Info and Resources

[Slack conversation](https://dsva.slack.com/archives/C010R6AUPHT/p1730822369440629)

Example template:
```
 ^ ##__ Submission details__
 ^
 ^ __Additional forms included__
 ^ ((forms_submitted))
 ^
 ^
```

`forms_submitted` and `files_submitted` should be within a code block but they are being excluded from the code block:
![Image](https://github.com/user-attachments/assets/1dae82f7-c97c-4fe3-a915-163617d56074)

It appears the `^` is not being applied to each bullet point added.

## Acceptance Criteria

- [ ] Lists submit inside a code block (denoted with `^`)  have the entire list inside a code block
- [ ] This work is added to the sprint review slide deck (key win bullet point and demo slide)

## QA Considerations

Permutations of code blocks and lists.

## Out of Scope
Updating Portal documentation. Collaboration with Portal is happening in parallel and the documentation will be updated after this is functional in the API.

**Comment by mjones-oddball** [comment]
@k-macmillan to add an AC regarding the API doc update needed

@kbelikova-oddball Can you find the corresponding Notify Exp ticket for the portal documentation updates? I think Kyle referred to it as advanced usage. Dave would like that ticket linked in the out of scope section so we know where that work is happening.

Additional note from Dave:
> In addition we'll want to think about change management/announcement of this functionality to existing customers if this is a feature others need/want beyond the current VA.gov request.

I think above should happen after the portal updates are made so folks have clear instruction on how to use this in the template and API request.

CC @davidconlon

**Comment by kalbfled** [comment]
This might work without additional changes after #169.  The newer version of Mistune handles block quotes the way I expect.
```
>>> import mistune
>>> print(mistune.html('> this\n>\n>\n> - a\n> - b'))
<blockquote>
    <p>this</p>
    <ul>
        <li>a</li>
        <li>b</li>
    </ul>
</blockquote>
```
(indentation added)

**Comment by kalbfled** [comment]
I have feedback from the HCD team.  Using template `a4b5df4f-6f29-4b0d-a61e-fb02e1ebdb40`, which contains this snippet of markdown . . .

```
^ ## Submission Details
^ __Form submitted__
^ Disability Claim (VA Form 21-526EZ)
^ 
^ __Additional forms included__
^ ((forms_submitted))
^ 
^ 
^  __Supporting evidence included__
^ For your privacy, we're showing only part of the names of the files you've submitted.
^ ((files_submitted))
^ 
^ 
^ __Date submitted__
^ ((date_submitted))
```
. . . and this personalization . . .
```
{
  :email_address=>"sam.stuckey@oddball.io",
  :template_id=>"a4b5df4f-6f29-4b0d-a61e-fb02e1ebdb40",
  :personalisation=> {
    :first_name=>"FAKEMAN_DELAYED",
    :date_submitted=>"November 5, 2024 8:14 p.m. UTC",
    :forms_submitted=> [
      "VA Form 21-4142",
      "VA Form 21-0781",
      "VA Form 21-8940"
    ],
    :files_submitted=> [
      "extXas.pdf",
      "extXas.pdf",
      "extXas.pdf"
    ],
    :date_of_failure=>"November 5, 2024 8:14 p.m. UTC"
  }
}
```
. . . I expect the output html to have the lists inside the block quote, like this:
```
<blockquote>
  <h2>Submission Details</h2>
  ...
  <p>For your privacy, we're showing only part of the names of the files you've submitted.</p>
  <ul>
    <li>file 1</li>
    <li>file 2</li>
    ...
  </ul>
  ...
</blockquote>
```
What actually happens is this:
![Image](https://github.com/user-attachments/assets/912c0fcf-36af-4c1d-ab2f-81d6270401d9)

The underlying html ends the blockquote before the start of the list.  The is probably a function of my regex for identifying block quotes.

**Comment by kalbfled** [comment]
This is taking a little longer than I hoped, but I'm getting there.  After tweaking the block quote regex that substitutes `>` for `^`, the problem is that the personalization substitutions are removing the block quote character in front of list items.  I have written unit tests that recreate problem reported by the user.

**Comment by kalbfled** [comment]
This has been more pernicious than I anticipated.  I might have a workable solution, but I have 4 failing unit tests which I will investigate in the morning.  If I don't have this ready for review by tomorrow, I'm going to mark it "off track" because QA work remains.

**Comment by cris-oddball** [comment]
PRs approved and merged, Deployed to Perf for testing.

**Comment by cris-oddball** [comment]
**QA TESTING IN PROGRESS**

- [ ] TC01 The template in Perf "05ef1670-fdcf-4cd8-b432-b457fcb0fdef", "QA Testing Carets", displays the blockquote correctly within the Portal, including nested lines in blockquotes.
- [ ] TC02 Sending an email with the above template and a list within the blockquote results in a delivered email that correctly displays the bulleted list and nested lines.
- [ ] TC03 Sending an email with the above template results  in a delivered email that correctly displays the line `This ^  is not a blockquote because the caret is not at the start.` noting that the up caret `^` will be replaced with a side caret `>` .
- [ ] TC04 Sending an email with the template XXXXX, XXXX results in a template that correctly displays all the formatting (did not introduce a regression.

**Comment by k-macmillan** [comment]
The PR associated with this work was reverted due to breaking changes in template rendering.

**Comment by cris-oddball** [comment]
@kalbfled Example of the issue seen during testing with the Portal UI:

![Image](https://github.com/user-attachments/assets/31d179b0-e472-4eea-ba2a-6b6f526c39f0)

Note that adding a bulleted list directly into a blockquote element in a template was also still broken. Continuation of this work should include the ability to insert a bullet list into a blockquote and not have it break the blockquote, not just a personalisation value that happens to include a bulleted list when sending.
```
^ This is a Blockquote
^ 
^ - bullet list 1
^ - bullet list 2
^ End Blockquote
```

**Comment by kalbfled** [comment]
This work is completely out of scope now.  It isn't obvious to me that the portal problems are not a portal bug, and explicit lists in block quotes is a new feature.

**Comment by npmartin-oddball** [comment]
Ticket being sent back to the Feature backlog. A new ticket is being created by @kalbfled re: unit testing and spec work. Discussion occurring between @k-macmillan and @nathanbwright re: next steps.

**Comment by cris-oddball** [comment]
NOTE:

This ticket is still necessary to display a list within a blockquote when that list is passed through a personalisation field.

Using template ID, 86820037-1374-4072-9f9f-e74f235170a1, "QA All The Formatting", when inserting a list into the personalisation field that is within a block quote, the list is rendered outside of that blockquote:

![Image](https://github.com/user-attachments/assets/f12fb32a-3038-473d-a32b-e13b55c3b0c3)

Note that a bulleted list added directly to the blockquote does render the bulleted list within the blockquote directly as shown in the above screenshot, so this must be validated to ensure that it still works.

Adding this to the description.
